File: test-automation-core/src/main/java/com/nortal/test/core/model/QueryCache.java
package com.nortal.test.core.model;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class QueryCache {
	private final Map<String, String> queryToIdentifier = new HashMap<>();

	public void add(String query, String resolvedIdentifier) {
		queryToIdentifier.put(query, resolvedIdentifier);
	}

	public Optional<String> get(String query) {
		return Optional.ofNullable(queryToIdentifier.get(query));
	}
}

File: test-automation-core-old/src/main/java/com/nortal/test/core/model/QueryCache.kt
package com.nortal.test.core.model

import java.util.*

class QueryCache {
    private val queryToIdentifier: MutableMap<String, String> = HashMap()
    fun add(query: String, resolvedIdentifier: String) {
        queryToIdentifier[query] = resolvedIdentifier
    }

    operator fun get(query: String): Optional<String> {
        return Optional.ofNullable(queryToIdentifier[query])
    }
}